lib/ssl/ssl3con.c

```diff
@@ -10371,15 +9664,14 @@ ssl3_HandleCertificateVerify(sslSocket *ss, SSL3Opaque *b, PRUint32 length,
         goto alert_loser;
     }
 
-    if (ss->ssl3.hs.hashType != handshake_hash_record) {
-        if (!hashes) {
-            PORT_Assert(0);
-            desc = internal_error;
-            errCode = SEC_ERROR_LIBRARY_FAILURE;
-            goto alert_loser;
-        }
-        hashesForVerify = hashes;
-    } else {
+    if (!hashes) {
```

Comment:
Not sure that I understand given that the ssl3.hs.ws check is above this, but it's a nit only, so do as you see fit.